internal/postprocessor/mainifest.go

```diff
@@ -0,0 +1,267 @@
+func (p *postProcessor) Manifest() (map[string]ManifestEntry, error) {
+	log.Println("updating gapic manifest")
+	entries := map[string]ManifestEntry{} // Key is the package name.
+	f, err := os.Create(filepath.Join(p.googleCloudDir, "internal", ".repo-metadata-full.json"))
+	if err != nil {
+		return nil, err
+	}
+	defer f.Close()
+	for _, manual := range manualEntries {
+		entries[manual.DistributionName] = manual
+	}
+	for inputDir, conf := range p.config.GoogleapisToImportPath {
+		yamlPath := filepath.Join(p.googleapisDir, inputDir, conf.ServiceConfig)
+		yamlFile, err := os.Open(yamlPath)
+		if err != nil {
+			return nil, err
+		}
+		yamlConfig := struct {
+			Title string `yaml:"title"` // We only need the title field.
+		}{}
+		if err := yaml.NewDecoder(yamlFile).Decode(&yamlConfig); err != nil {
+			return nil, fmt.Errorf("decode: %v", err)
+		}
+		docURL, err := docURL(p.googleCloudDir, conf.ImportPath)
+		if err != nil {
+			return nil, fmt.Errorf("unable to build docs URL: %v", err)
+		}
+		// TODO(codyoss): We can read a files doc.go to see if it has a warning
+		// if needs. For beta/alpha use a guess
+		releaseLevel, err := releaseLevel(p.googleCloudDir, conf.ImportPath)
+		if err != nil {
+			return nil, fmt.Errorf("unable to calculate release level for: %v", inputDir)
+		}
+
+		entry := ManifestEntry{
+			DistributionName:  conf.ImportPath,
+			Description:       yamlConfig.Title,
+			Language:          "Go",
+			ClientLibraryType: "generated",
+			DocsURL:           docURL,
+			ReleaseLevel:      releaseLevel,
+			LibraryType:       GapicAutoLibraryType,
+		}
+		entries[conf.ImportPath] = entry
+	}
+	enc := json.NewEncoder(f)
+	enc.SetIndent("", "  ")
+	return entries, enc.Encode(entries)
+}
+
+func docURL(cloudDir, importPath string) (string, error) {
+	suffix := strings.TrimPrefix(importPath, "cloud.google.com/go/")
+	mod, err := gocmd.CurrentMod(filepath.Join(cloudDir, suffix))
+	if err != nil {
+		return "", err
+	}
+	pkgPath := strings.TrimPrefix(strings.TrimPrefix(importPath, mod), "/")
+	return "https://cloud.google.com/go/docs/reference/" + mod + "/latest/" + pkgPath, nil
+}
+
+func releaseLevel(cloudDir, importPath string) (string, error) {
+	i := strings.LastIndex(importPath, "/")
+	lastElm := importPath[i+1:]
+	if strings.Contains(lastElm, "alpha") {
+		return "alpha", nil
+	} else if strings.Contains(lastElm, "beta") {
+		return "beta", nil
+	}
+
+	// Determine by scanning doc.go for our beta disclaimer
+	gapicRelPath := strings.TrimPrefix(importPath, "cloud.google.com/go/")
+	docFile := filepath.Join(cloudDir, gapicRelPath, "doc.go")
+	f, err := os.Open(docFile)
+	if err != nil {
+		return "", err
+	}
+	defer f.Close()
+
+	scanner := bufio.NewScanner(f)
+	var lineCnt int
+	for scanner.Scan() && lineCnt < 50 {
+		line := scanner.Text()
+		if strings.Contains(line, betaIndicator) {
+			return "beta", nil
+		}
+	}
+	return "ga", nil
+}
+
+var manualEntries = []ManifestEntry{
```

Comment:
I know these are very stable, but just have to ask: Should this be extracted to a config file too?

Reply:
I like the idea, done